menu/cbs/menu_cbs_ok.c

```diff
@@ -2333,11 +2341,44 @@ static bool playlist_entry_path_is_valid(const char *entry_path)
    return false;
 }
 
+#ifdef HAVE_ENTRYSTATES
+static bool playlist_entry_state_is_valid(const char *entry_state)
+{
+   char *file_path     = NULL;
+
+   if (string_is_empty(entry_state))
+      return true;
+
+   file_path = strdup(entry_state);
```

Comment:
You don't need to `strdup()` the path here - that was only required in the function you copy/pasted from :)

In fact, I don't think this function is needed at all (see next comment)